apps/storefront/src/pages/OrderDetail/components/OrderAction.tsx

```diff
@@ -285,8 +285,19 @@ export default function OrderAction(props: OrderActionProps) {
     orderId,
     ipStatus = 0,
     invoiceId,
+    poNumber,
   } = detailsData;
 
+  const getPaymentMessage = () => {
+    const message = poNumber ? 'orderDetail.paidWithPo' : 'orderDetail.paidInFull';
+    return (
+      createAt &&
+      b3Lang(message, {
+        paidDate: displayFormat(createAt, true),
+      })
+    );
+  };
+
```

Comment:
As you're using a prop you should use `useCallback`
```suggestion
  const getPaymentMessage = useCallback(() => {
    const message = poNumber ? 'orderDetail.paidWithPo' : 'orderDetail.paidInFull';
    let message = ''
    
    if(createdAt) {
      if(poNumber) {
        message = b3Lang('orderDetail.paidWithPo', {
          paidDate: displayFormat(createAt, true),
        })
      } else {
        message = b3Lang('orderDetail.paidInFull', {
          paidDate: displayFormat(createAt, true),
        })
      }
    }

    return message
  }, [poNumber, createAt]);

```

Reply:
👍